lib/xlat_tables_v2/xlat_tables_internal.c

```diff
@@ -786,6 +786,19 @@ void mmap_add_region_ctx(xlat_ctx_t *ctx, const mmap_region_t *mm)
 	       && (mm_cursor->size < mm->size))
 		++mm_cursor;
 
+	/*
+	 * Find the last entry marker in the mmap
+	 */
+	mm_last = ctx->mmap;
+	while ((mm_last->size != 0U) && (mm_last < mm_end)) {
+		++mm_last;
+	}
+
+	if (mm_last->size != 0U) {
```

Comment:
Actually, the function `mmap_add_region_check` checks that there is enough space in the mmap array. This should be an assert.